x-pack/test/functional/services/ml/data_visualizer_index_based.ts

```diff
@@ -37,12 +37,88 @@ export function MachineLearningDataVisualizerIndexBasedProvider(
       await this.assertTotalDocumentCount(expectedFormattedTotalDocCount);
     },
 
-    async assertFieldsPanelsExist(expectedPanelCount: number) {
-      const allPanels = await testSubjects.findAll('~mlDataVisualizerFieldsPanel');
-      expect(allPanels).to.have.length(
-        expectedPanelCount,
-        `Expected field panels count to be '${expectedPanelCount}' (got '${allPanels.length}')`
-      );
+    async assertDataVisualizerTotalDocCountHeaderExist() {
+      await testSubjects.existOrFail(`mlDataVisualizerTotalDocCountHeader`);
+    },
+
+    async assertDataVisualizerTotalDocCountChartExist() {
+      await testSubjects.existOrFail(`mlFieldDataCardDocumentCountChart`);
+    },
+
+    async assertDataVisualizerSearchPanelExist() {
+      await testSubjects.existOrFail(`mlDataVisualizerSearchPanel`);
+    },
+
+    async assertDataVisualizerSearchQueryInputExist() {
+      await testSubjects.existOrFail(`mlDataVisualizerQueryInput`);
+    },
+
+    async assertDataVisualizerFieldCountPanelExist() {
+      await testSubjects.existOrFail(`mlDataVisualizerFieldCountPanel`);
+    },
+
+    async assertDataVisualizerMetricFieldsSummaryExist() {
+      await testSubjects.existOrFail(`mlDataVisualizerMetricFieldsSummary`);
+    },
+
+    async assertVisibleMetricFieldsCount(count: number) {
+      const expectedCount = count.toString();
+      await testSubjects.existOrFail('mlDataVisualizerVisibleMetricFieldsCount');
+      await retry.tryForTime(5000, async () => {
+        const actualCount = await testSubjects.getVisibleText(
+          'mlDataVisualizerVisibleMetricFieldsCount'
+        );
+        expect(expectedCount).to.eql(
+          expectedCount,
+          `Expected metric fields count count to be '${expectedCount}' (got '${actualCount}')`
+        );
+      });
+    },
+
+    async assertMetricFieldsCount(count: number) {
```

Comment:
How is this one different from `assertVisibleMetricFieldsCount`?

Reply:
I see that you've updated the method name to clarify the difference between `assertVisibleMetricFieldsCount` and `assertTotalMetricFieldsCount`. But as far as I can see, the implementation is still the same, as both methods are targeting the `mlDataVisualizerVisibleMetricFieldsCount` test subject.

Reply:
You're right I missed that - updated here https://github.com/elastic/kibana/pull/85726/commits/46d06b5dfc7368ad61d09511ef80511ba8c37943